packages/flutter_tools/lib/src/test/coverage_collector.dart

```diff
@@ -237,91 +236,114 @@ Future<FlutterVmService> _defaultConnect(Uri serviceUri) {
       serviceUri, compression: CompressionOptions.compressionOff, logger: globals.logger,);
 }
 
-Future<Map<String, dynamic>> collect(Uri serviceUri, bool Function(String) libraryPredicate, {
+Future<Map<String, dynamic>> collect(Uri serviceUri, Set<String> libraryNames, {
   bool waitPaused = false,
   String debugName,
   Future<FlutterVmService> Function(Uri) connector = _defaultConnect,
   @visibleForTesting bool forceSequential = false,
 }) async {
   final FlutterVmService vmService = await connector(serviceUri);
-  final Map<String, dynamic> result = await _getAllCoverage(vmService.service, libraryPredicate, forceSequential);
+  final Map<String, dynamic> result = await _getAllCoverage(vmService.service, libraryNames, forceSequential);
   await vmService.dispose();
   return result;
 }
 
+bool _versionCheck(vm_service.Version version, int minMajor, int minMinor) {
+  return version.major > minMajor || (version.major == minMajor && version.minor >= minMinor);
+}
+
 Future<Map<String, dynamic>> _getAllCoverage(
   vm_service.VmService service,
-  bool Function(String) libraryPredicate,
+  Set<String> libraryNames,
   bool forceSequential,
 ) async {
   final vm_service.Version version = await service.getVersion();
-  final bool reportLines = (version.major == 3 && version.minor >= 51) || version.major > 3;
+  final reportLines = _versionCheck(version, 3, 51);
```

Comment:
A given flutter tool will only ever build with a single version of Dart. Do you know when this will roll into the framework? because if we've past that point it would be better to remove the version check so we can only have a single implementation of this.

Reply:
libraryFilters is not in VM version 2.17, but reportLines definitely is, so I can remove that flag.